### ISSUE TITLE
Support GitHub Deployments 'task' parameter

### ISSUE THREAD

**Issue by fernandrone** [woodpecker-ci/woodpecker#3445]
### Clear and concise description of the problem

As a user of Woodpecker I want the `deployment` event to support the `task` parameter. I'm open to submitting a pull request if this is a desired feature and we can agree on a design.

The `deployment` event is [a Github-only event](https://woodpecker-ci.org/docs/administration/forges/overview#supported-features) that can currently be filtered by an [environment](https://woodpecker-ci.org/docs/usage/workflow-syntax#environment-1). However the [GitHub API](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment) also accepts a `task` parameter which has the intention of further "filtering" the execution path of a pipeline:

> The `task` parameter is used by the deployment system to allow different execution paths. In the web world this might be `deploy:migrations` to run schema changes on the system. In the compiled world this could be a flag to compile an application with debugging enabled.

At the moment, Woodpecker does not parse this parameter, so it is discarded. However, if it were to be parsed, it would allow greater flexibility when using the deployment event.

For a more specific use case, we've adopted Woodpecker and we're using GitHub Deployments to trigger pipelines on demand. Although the `manual` event can be used we like many of the features that come out of the box with GitHub deployments (such as deployment queues as well as a page on your repository with all the past deployments, sorted by environment). The extra variable would allow more fine-grained control, that today can only be achieved with workarounds such as custom branch prefixes or tag names; for example, to build only one specific module, or only run the "migration" step.

### Suggested solution

For example, the proposed syntax below could be used to run only database migrations in the production environment.

```
when:
  - event: deployment
     environment: production
     task: migration
```

Or, in a monorepo with multiple applications, deploy a single application.

```yaml
when:
  - event: deployment
     environment: staging
     task: deploy:app1
```

A new environment variable would likely be required, seeing that `CI_PIPELINE_DEPLOY_TARGET` refers to the value of environment I think a `CI_PIPELINE_DEPLOY_TASK` would likely be a good candidate (and a `CI_PREV_PIPELINE_DEPLOY_TASK` as well).

From what I can see the implementation should be straightforward, as it's an existing field on a GitHub event, you could add a `hook.GetDeployment().GetTask()` on parseDeployHook: https://github.com/woodpecker-ci/woodpecker/blob/main/server/forge/github/parse.go#L122-L133

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by fernandrone** [comment]
I've started to work on this here in this draft https://github.com/woodpecker-ci/woodpecker/pull/3512 (I still haven't tested it, but sharing it here in case anyone wants to give some early feedback).

**Comment by fernandrone** [comment]
>  Not fixed in core yet. Please don't use issue linking from other repos if you would like to get this in the main repo

Sorry about that. We're keeping a fork of woodpecker with the version that we deploy to our internal infrastructure. I didn't realize that if we linked to this issue from our fork, GitHub would automatically close it when our fork was updated (seems like a very silly for GitHub to do).

I'll refrain from issue linking in the future... our might make our fork private to avoid spamming the main repo with unneeded notifications. I'll see that works best :+1:
